File: ClasesSergio/codigo/src/clase7/ej3.java
package clase7;

public class ej3 {

	public static int[][] sumarMatrices(int[][] matriz1, int[][] matriz2) {
        // Verificar si las matrices tienen las mismas dimensiones
        if (matriz1.length != matriz2.length || matriz1[0].length != matriz2[0].length) {
            System.out.println("Las matrices no tienen las mismas dimensiones.");
            return null;
        }

        int[][] matrizSuma = new int[matriz1.length][matriz1[0].length];

        for (int i = 0; i < matriz1.length; i++) {
            for (int j = 0; j < matriz1[0].length; j++) {
                matrizSuma[i][j] = matriz1[i][j] + matriz2[i][j];
            }
        }
        return matrizSuma;
    }

    public static void imprimirMatriz(int[][] matriz) {
        for (int i = 0; i < matriz.length; i++) {
            for (int j = 0; j < matriz[0].length; j++) {
                System.out.print(matriz[i][j] + " ");
            }
            System.out.println();
        }
    }
    
	public static void main(String[] args) {
        int[][] matriz1 = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
        int[][] matriz2 = {{9, 8, 7}, {6, 5, 4}, {3, 2, 1}};

        int[][] matrizSuma = sumarMatrices(matriz1, matriz2);

        imprimirMatriz(matrizSuma);
    }

}

File: ClasesNacho/Clase12/ej3.py
def son_matrices_iguales(matriz1, matriz2):
    # Verificar si ambas matrices tienen el mismo número de filas
    if len(matriz1) != len(matriz2):
        return False
    
    for i in range(len(matriz1)):
        if len(matriz1[i]) != len(matriz2[i]):
            return False
    
    for i in range(len(matriz1)):
        for j in range(len(matriz1[i])):
            if matriz1[i][j] != matriz2[i][j]:
                return False
    
    return True

def escribir_matriz(matriz):
    for fila in matriz:
        for elemento in fila:
            print(elemento, end=" ")
        print()

# Ejemplo de uso
matriz_a = [
    [1, 2, 3],
    [4, 5, 6]
]

matriz_b = [
    [1, 2, 3],
    [4, 5, 6]
]

matriz_c = [
    [1, 2, 3],
    [4, 5, 7]
]

print("Matriz 1: ")
escribir_matriz(matriz_a)

print("Matriz 2: ")
escribir_matriz(matriz_b)

print("Matriz 3: ")
escribir_matriz(matriz_c)

if son_matrices_iguales(matriz_a, matriz_b):
    print("Las matrices 1 y 2 son iguales.")
else:
    print("Las matrices 1 y 2 no son iguales.")

if son_matrices_iguales(matriz_a, matriz_c):
    print("Las matrices 1 y 3 son iguales.")
else:
    print("Las matrices 1 y 3 no son iguales.")
